src/Google.Api.Gax/CallSettings.cs

```diff
@@ -62,6 +73,65 @@ public sealed class CallSettings
             WriteOptions = WriteOptions,
             PropagationToken = PropagationToken,
             Credentials = Credentials,
+            RetrySettings = RetrySettings?.Clone(),
         };
+
+        // Merge other into this.
+        // Other is unchanged. Returns this
+        // Internal? Or Public?
+        public CallSettings Merge(CallSettings other)
+        {
+            if (other == null)
+            {
+                return this;
+            }
+            if (other.Headers != null)
```

Comment:
Yes, we did discuss this.
However (aside from the question of consistency with other merging properties), that would mean it's impossible to remove headers during an override, which I feel uncomfortable about.
I'll put a TODO in, and we should probably discuss this further and/or see what other code does in similar situations.